### PR TITLE
add `metrics` feature

### PR DESCRIPTION
**Pull request by wngr**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -112,12 +124,42 @@ impl<K: Hash + Eq, V: Weighable> Default for WeightCache<K, V> {
     }
 }
 
+#[cfg(feature = "metrics")]
+struct Metrics {
+    hits: GenericCounter<AtomicU64>,
```
Comment (rklaehn) [inline src/lib.rs:130]: Is this a re-export of std AtomicU64, or its own thing?

File: src/lib.rs [hunk 2]
```diff
@@ -152,6 +214,8 @@ impl<K: Hash + Eq, V: Weighable, S: BuildHasher> WeightCache<K, V, S> {
     pub fn put(&mut self, key: K, value: V) -> Result<(), ValueTooBigError> {
         let weight = V::measure(&value);
         if weight > self.max {
+            #[cfg(feature = "metrics")]
+            self.metrics.inserts_fail.inc();
```
Comment (rklaehn) [inline src/lib.rs:260]: Not sure we need this. It will always be 0 for a normal program, and you do get an error.

**Review by rklaehn — APPROVED** [review]
Looks pretty straightforward.

Sorry about the stupid question, but what would happen if you had multiple caches? Would that then be multiple registries, or somehow combined? Asking because of `self.metrics.size.set(0);`, among other things.

File: src/lib.rs (hunk 1 above)
Comment (wngr) [reply src/lib.rs:130]: It's their wrapper around it providing some trait impl they need.

File: src/lib.rs (hunk 2 above)
Comment (wngr) [reply src/lib.rs:260]: Well, it shows you that the cache is not correctly sized for your data, I think that's valuable.

File: src/lib.rs (hunk 1 above)
Comment (rklaehn) [reply src/lib.rs:130]: ok, so the usual rust newtype boilerplate...

**Pull request merged**